Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1

```diff
@@ -58,560 +60,622 @@ $MsDeployInstallPathRegKey = "HKLM:\SOFTWARE\Microsoft\IIS Extensions\MSDeploy"
-    -addBinding $addBinding -protocol $protocol -ipAddress $ipAddress -port $port -hostname $hostname -assignDupBindings $assignDupBindings
+	param(
+		[string]$siteName,
+		[string]$appPoolName,
+		[string]$physicalPath,
+		[string]$authType,
+		[string]$userName,
+		[string]$password,
+		[string]$addBinding,
+		[string]$protocol,
+		[string]$ipAddress,
+		[string]$port,
+		[string]$hostname,
+		[string]$assignDupBindings
+	)
+
+	$doesWebSiteExists = Does-WebSiteExists -siteName $siteName
+	
+	if( -not $doesWebSiteExists)
+	{
+		Create-WebSite -siteName $siteName -physicalPath $physicalPath
+	}
+
+	Update-WebSite -siteName $siteName -appPoolName $appPoolName -physicalPath $physicalPath -authType $authType -userName $userName -password $password `
+	-addBinding $addBinding -protocol $protocol -ipAddress $ipAddress -port $port -hostname $hostname -assignDupBindings $assignDupBindings
 }
 
 function Create-And-Update-AppPool
 {
-    param(
-        [string]$appPoolName,
-        [string]$clrVerion,
-        [string]$pipeLineMode,
-        [string]$identity,
-        [string]$userName,
-        [string]$password
-    )
-
-    $doesAppPoolExists = Does-AppPoolExists -appPoolName $appPoolName
-
-    if(-not $doesAppPoolExists)
-    {
-        Create-AppPool -appPoolName $appPoolName
-    }
-
-    Update-AppPool -appPoolName $appPoolName -clrVersion $clrVerion -pipeLineMode $pipeLineMode -identity $identity -userName $userName -password $password
+	param(
+		[string]$appPoolName,
+		[string]$clrVerion,
+		[string]$pipeLineMode,
+		[string]$identity,
+		[string]$userName,
+		[string]$password
+	)
+
+	$doesAppPoolExists = Does-AppPoolExists -appPoolName $appPoolName
+
+	if(-not $doesAppPoolExists)
+	{
+		Create-AppPool -appPoolName $appPoolName
+	}
+
+	Update-AppPool -appPoolName $appPoolName -clrVersion $clrVerion -pipeLineMode $pipeLineMode -identity $identity -userName $userName -password $password
+}
+
+function Decode-String
+{
+	param(
+		[string]$string
+	)
+	if(-not ([string]::IsNullOrEmpty($string)))
+	{
+		$decodedString = [System.Text.Encoding]::Unicode.GetString([System.Convert]::FromBase64String($string));
+		return $decodedString
+	}
+	return [string]::Empty
+}
+
+function Get-OverRideParameters
+{
+	param(
+		[string]$overRideParameters
+	)
+
+	$overRideParameters = Decode-String -String $overRideParameters.Trim('"')
+
+	$overRideArgs = [string]::Empty
+
+	[array]$overRideParams = $overRideParameters.Split([System.Environment]::NewLine, [System.StringSplitOptions]::RemoveEmptyEntries)
+
+	foreach($overRideParam in $overRideParams)
+	{
+		if(-not [string]::IsNullOrEmpty($overRideParam.Trim(' ')))
+		{
+			if($overRideParam -notmatch "Connection String" -and $overRideParam -notmatch "ConnectionString")
+			{
+				[array]$splitOverRide  = $overRideParam.Split('=')
+				$name = $splitOverRide[0].Trim('"').Trim("'").Trim(' ')
+				$value = $splitOverRide[1].Trim('"').Trim("'").Trim(' ')
+				
+				$cleanOverRide = "`"$name`"=`"$value`""
```

Comment:
We are aware of an issue with complex msdeploy params and the issue we are planning to fix in different layer. Currently if that issue is blocking you can use this work around in your repo, but we will fix that in remote deployer layer not in task and then we will remove hack of putting single in place of double quotes.